### ISSUE TITLE
URL hint is unclear while reporting an issue

### ISSUE THREAD

**Issue by MrunalrajRedij** [OWASP/BLT-Flutter#364]
In report screen, in a first textfield it gives hint as `App name / URL`.  So I typed as it is,  app name after back slash the url of an app `example: xyz / https:/www.xyz.com `
But report form only accepts a valid URL as an input. Otherwise there is will be an exception returning `Enter valid URL`

I think #326  issue is also related to it because while creating an issue I got that `URL exception` msg and I immediately thought it was an image uploading issue. Turns out images were working fine, single as well as multiple. But It was me who was entering incorrect URL.

Can we give more clear hints? or a separate textfields for app name and URL?

**Comment by Dante291** [comment]
Hey i think you got it all wrong, it is basically either type the name of the app or its URL.

**Comment by MrunalrajRedij** [comment]
I looked at a code. 
report_bug.dart: line 910

```
Issue issue = Issue(
                      ...
                      url: _titleController.text,
                      title: _titleController.text,
                      ...
                    );
```
So even if we input only app name, we are assigning it to the url field. There is no check happening for valid URL and condition to assign it _url_ field if _valid URL_.
And because of than server is returning the _error_ because it is not a valid URL.

**Comment by MrunalrajRedij** [comment]
That's what I am asking.
So it's compulsory to give include _https://_ ?
And one should not input only app name?

**Comment by Dante291** [comment]
yes for valid URL you need to add https:// also for app name i guess we need to correct it as currently we are using only URL to identify on what platform bug is occuring.

**Comment by MrunalrajRedij** [comment]
> bro add http:// too in the url and it will work fine

Sorry, I think I am not describing my issue properly.
Looking at hint given, it looks like I can type either app name or URL as you too perviously mentioned. 

But inputting _app name_ alone will give an _Invalid URL_ error from server.
I don't think this is intended. Is it?

**Comment by Dante291** [comment]
Yeah gotta fix it, it should be only URL and onFocus hint text should be an example URL starting with https://

do you wanna work on this? or should i fix this?

**Comment by MrunalrajRedij** [comment]
> yes for valid URL you need to add https:// also for app name i guess we need to correct it as currently we are using only URL to identify on what platform bug is occuring.

Okay, got it so URL is mandatory.
I mean it's not too big on an issue but should the hint be remain as it is or changed? 
Also can we make two separate fields for app name and URL? I am guessing it will also require changing backend too.

**Comment by MrunalrajRedij** [comment]
I wanna work on this

**Comment by Dante291** [comment]
> > yes for valid URL you need to add https:// also for app name i guess we need to correct it as currently we are using only URL to identify on what platform bug is occuring.
> 
> Okay, got it so URL is mandatory. I mean it's not too big on an issue but should the hint be remain as it is or changed? Also can we make two separate fields for app name and URL? I am guessing it will also require changing backend too.

For now fix this only

**Comment by Dante291** [comment]
@DonnieBLT Close this issue as it is completed.
